### PR TITLE
Adding Microsoft.Private.CoreFx.UAP package build

### PR DESCRIPTION
**Pull request by joperezr**
Build UAP flat package called `Microsoft.Private.CoreFx.UAP`

cc: @weshaggard @danmosemsft @ericstj @jkotas

### CODE REVIEW

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+        <Original>%(_fileExisting.Identity)</Original>
+      </_fileNames>
+      <_filesFiltered Include="@(_fileNames->'%(Original)')" />
+    </ItemGroup>
+
+    <VerifyClosure Sources="@(_filesFiltered)"
+                   IgnoredReferences="@(IgnoredReference)" />
+  </Target>
+
+  <Target Name="VerifyNETStandard" AfterTargets="Build">
```
Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: I guess these targets can get refactored into a dir.targets so they get used by several pkgprojs

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+    </NativeFile>
+    <File Include="@(NativeFile)" />
+    <!-- force a missing file if native build is absent -->
+    <File Include="$(NativeBinDir)/MISSING_NATIVE_BUILD" Condition="'@(NativeFile)' == ''" />
+  </ItemGroup>
+
+  <ItemGroup>
+    <IgnoredReference Include="System.Private.CoreLib" />
+    <IgnoredReference Include="Windows" />
+    <IgnoredReference Include="Microsoft.Win32.Registry" /> <!-- Ignore for now since it's being exclude from closure bellow -->
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: Can you please make sure there are bugs tracking each of these issues and add a reference to the issue in the comment?

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+    <IgnoredReference Include="System.IO.IsolatedStorage" /> <!-- Ignore for now since it's being exclude from closure bellow -->
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: nit: bellow -> below

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+    <TargetFrameworkName>uap</TargetFrameworkName>
+    <TargetFrameworkVersion>10.1</TargetFrameworkVersion>
+    <TargetFramework>$(TargetFrameworkName)$(TargetFrameworkVersion)</TargetFramework>
+    <SkipPackageFileCheck>true</SkipPackageFileCheck>
+    <SkipValidatePackage>true</SkipValidatePackage>
+
+    <RefBinDir>$(UAPPackageRefPath)</RefBinDir>
+    <LibBinDir>$(UAPPackageRuntimePath)</LibBinDir>
+
+    <NETStandardLibraryPackage>NETStandard.Library2</NETStandardLibraryPackage>
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: Can you remove these and use the versions that are coming from dependencies.props? We shouldn't beusing NETSTandard.Library2 any more.

File: pkg/descriptions.json [hunk 5]
```diff
@@ -116,6 +116,11 @@
     "CommonTypes": []
   },
   {
+    "Name": "Microsoft.Private.CoreFx.UAP",
+    "Description": "Package used to represent the portions of NETCoreApp that come from CoreFx for UAP.",
```
Comment (jkotas) [inline pkg/descriptions.json]: This should not be talking about NetCoreApp, right? I think it should say:

`Package used to represent the portions of UAP that come from CoreFx.`

File: pkg/Microsoft.Private.CoreFx.UAP/Configurations.props [hunk 6]
```diff
@@ -0,0 +1,8 @@
+﻿<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  <PropertyGroup>
+    <BuildConfigurations>
+      uap-Windows_NT;
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Configurations.props]: I assume you are going to have this package contain both the uap and uap-aot assets?

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.props [hunk 7]
```diff
@@ -0,0 +1,54 @@
+<?xml version="1.0" encoding="utf-8"?>
+<Project ToolsVersion="14.0" DefaultTargets="Build" xmlns="http://schemas.microsoft.com/developer/msbuild/2003">
+  
+  <PropertyGroup>
+    <_runtimeOSVersionIndex>$(RuntimeOS.IndexOfAny(".-0123456789"))</_runtimeOSVersionIndex>
+    <_runtimeOSFamily Condition="'$(_runtimeOSVersionIndex)' != '-1'">$(RuntimeOS.SubString(0, $(_runtimeOSVersionIndex)))</_runtimeOSFamily>
+  </PropertyGroup>
+  
+  <Choose>
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.props]: What about the aot RID?

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj (hunk 2 above)
Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: Done.

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj (hunk 3 above)
Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: Fixed.

File: pkg/descriptions.json (hunk 5 above)
Comment (joperezr) [inline pkg/descriptions.json]: changed.

File: pkg/Microsoft.Private.CoreFx.UAP/Configurations.props (hunk 6 above)
Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Configurations.props]: Yes, the plan is to build:
 - 1 identity package
 - 4 uap runtime - specific packages (x86, x64, arm, arm64)
 - 4 uapaot runtime - specific packages (x86-aot, x64-aot, arm-aot, arm64-aot)

This PR takes care of the first two bullet points, and I will do a separate PR for the third one.

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.props (hunk 7 above)
Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.props]: I was planning on doing the aot RID on a subsequent PR

**Comment by joperezr** [comment]
@weshaggard note that with 824f666 I'm adding more "hacks" regarding the calculation of pkg paths, which we did discuss that should go into a separate issue as binplacing might need some re-engineering in order for that to work. I'll open an issue to keep track of this.

File: pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj [hunk 8]
```diff
@@ -0,0 +1,112 @@
...
+    <TargetFrameworkVersion>10.1</TargetFrameworkVersion>
+    <TargetFramework>$(TargetFrameworkName)$(TargetFrameworkVersion)</TargetFramework>
+    <SkipPackageFileCheck>true</SkipPackageFileCheck>
+    <SkipValidatePackage>true</SkipValidatePackage>
+
+    <RefBinDir>$(UAPPackageRefPath)</RefBinDir>
+    <LibBinDir Condition="'$(_bc_TargetGroup)'=='uap'">$(UAPPackageRuntimePath)</LibBinDir>
+    <LibBinDir Condition="'$(_bc_TargetGroup)'=='uapaot'">$(UAPAOTPackageRuntimePath)</LibBinDir>
+
+    <NETStandardLibraryPackage>NETStandard.Library2</NETStandardLibraryPackage>
```
Comment (weshaggard) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: We should still try to share these package properties with dependencies.props and NETStandard.Library2 shouldn't be used any more.

Comment (joperezr) [inline pkg/Microsoft.Private.CoreFx.UAP/Microsoft.Private.CoreFx.UAP.pkgproj]: Fixed.

**Pull request merged**